Sources/JWTKit/Signing/JWTSigners.swift

```diff
@@ -6,12 +6,16 @@ import struct Foundation.Data
 public final class JWTSigners {
     /// Internal storage.
     private var storage: [JWKIdentifier: JWTSigner]
+    
+    /// Internal storage for RSA keys for JWK with unknown algorithm.
+    private var rsaKeysStorage: [JWKIdentifier: RSAKey]
```

Comment:
If an arbitrary RSA key can work with rs256, rs384, and rs512, then I think we should refactor the RSA JWTSigner to be able to infer the algorithm. Instead of being locked to a single one, we could allow users to pass a "defaultAlgorithm" in case no `alg` header is present in the JWT. If the header is present we would choose the specific algorithm based on that. Does that make sense? In other words, I don't think we need to modify `JWTSigners` here. We should be able to do this by making the RSA JWTSigner more powerful.

Reply:
I think the order should be: `JWK` (we should trust keys from provider) -> `JWT` (`JOSE` header) -> exception, so when there is no algorithm in JWK and JWT we will throw an exception because that situation is very strange. We shouldn't specify default algorithm, because when provider changes the algorithm developers will have to rebuild their applications (and also different keys can use different algorithms).

The option that we will store RSAKeys instead JWTSigners and use RSAKey+alg to get appropriate signer is very good. It's much bigger change thus I will be also happy if you implement this.